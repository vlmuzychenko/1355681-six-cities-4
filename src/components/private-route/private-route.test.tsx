import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from 'react-router-dom';
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space.js";
import PrivateRoute from "./private-route";

const mockStore = configureStore([]);

const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  OFFER: `/offer`,
  FAVORITES: `/favorites`,
};

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

it(`Shoud PrivateRoute render correctly`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      authorizationData: {},
    }
  });

  const render = jest.fn();

  const tree = renderer
    .create(
        <BrowserRouter>
          <Provider store={store}>
            <PrivateRoute
              render={render}
              path={AppRoute.FAVORITES}
              exact={true}
              authStatus={AuthorizationStatus.AUTH} />
          </Provider>
        </BrowserRouter>
    )
        .toJSON();

  expect(tree).toMatchSnapshot();
});
