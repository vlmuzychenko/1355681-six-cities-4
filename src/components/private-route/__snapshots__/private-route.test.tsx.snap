// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoud PrivateRoute render correctly 1`] = `null`;
